Vite 4 support?

Having trouble installing it with Vite 4. Error below:

```
$ npm i vite-ssr -D
npm ERR! code ERESOLVE
npm ERR! ERESOLVE unable to resolve dependency tree
npm ERR! 
npm ERR! While resolving: starter@0.0.0
npm ERR! Found: vite@4.0.4
npm ERR! node_modules/vite
npm ERR!   dev vite@"^4.0.4" from the root project
npm ERR!   peer vite@"^4.0.0" from @vitejs/plugin-vue@4.0.0
npm ERR!   node_modules/@vitejs/plugin-vue
npm ERR!     dev @vitejs/plugin-vue@"^4.0.0" from the root project
npm ERR! 
npm ERR! Could not resolve dependency:
npm ERR! peer vite@"^2.7.7" from vite-ssr@0.16.0
npm ERR! node_modules/vite-ssr
npm ERR!   dev vite-ssr@"*" from the root project
npm ERR! 
npm ERR! Fix the upstream dependency conflict, or retry
npm ERR! this command with --force or --legacy-peer-deps
npm ERR! to accept an incorrect (and potentially broken) dependency resolution.
npm ERR! 
npm ERR! 
npm ERR! For a full report see:
npm ERR! /home/lau/.npm/_logs/2023-01-15T10_09_05_724Z-eresolve-report.txt

npm ERR! A complete log of this run can be found in:
npm ERR!     /home/lau/.npm/_logs/2023-01-15T10_09_05_724Z-debug-0.log

```

If I forced it:

```
$ npm i vite-ssr -D --force
npm WARN using --force Recommended protections disabled.
npm WARN ERESOLVE overriding peer dependency
npm WARN While resolving: vite-ssr@0.16.0
npm WARN Found: @vitejs/plugin-vue@4.0.0
npm WARN node_modules/@vitejs/plugin-vue
npm WARN   dev @vitejs/plugin-vue@"^4.0.0" from the root project
npm WARN 
npm WARN Could not resolve dependency:
npm WARN peerOptional @vitejs/plugin-vue@"^2.0.1" from vite-ssr@0.16.0
npm WARN node_modules/vite-ssr
npm WARN   dev vite-ssr@"*" from the root project
npm WARN 
npm WARN Conflicting peer dependency: @vitejs/plugin-vue@2.3.4
npm WARN node_modules/@vitejs/plugin-vue
npm WARN   peerOptional @vitejs/plugin-vue@"^2.0.1" from vite-ssr@0.16.0
npm WARN   node_modules/vite-ssr
npm WARN     dev vite-ssr@"*" from the root project
npm WARN ERESOLVE overriding peer dependency
npm WARN While resolving: vite-ssr@0.16.0
npm WARN Found: vite@4.0.4
npm WARN node_modules/vite
npm WARN   peer vite@"^4.0.0" from @vitejs/plugin-vue@4.0.0
npm WARN   node_modules/@vitejs/plugin-vue
npm WARN     dev @vitejs/plugin-vue@"^4.0.0" from the root project
npm WARN   2 more (vite-plugin-pages, the root project)
npm WARN 
npm WARN Could not resolve dependency:
npm WARN peer vite@"^2.7.7" from vite-ssr@0.16.0
npm WARN node_modules/vite-ssr
npm WARN   dev vite-ssr@"*" from the root project
npm WARN 
npm WARN Conflicting peer dependency: vite@2.9.15
npm WARN node_modules/vite
npm WARN   peer vite@"^2.7.7" from vite-ssr@0.16.0
npm WARN   node_modules/vite-ssr
npm WARN     dev vite-ssr@"*" from the root project

added 30 packages, and audited 224 packages in 9s
```


Thanks

--- a/examples/react-apollo/package.json
+++ b/examples/react-apollo/package.json
@@ -17,7 +17,7 @@
     "styled-components": "^5.3.0"
   },
   "devDependencies": {
-    "@vitejs/plugin-react": "^1.1.3",
-    "vite": "2.9.x"
+    "@vitejs/plugin-react": "^3.0.1",
+    "vite": "4.0.x"
   }
 }

--- a/examples/react/package.json
+++ b/examples/react/package.json
@@ -14,7 +14,7 @@
     "react-router-dom": "^6.2.2"
   },
   "devDependencies": {
-    "@vitejs/plugin-react": "^1.1.3",
-    "vite": "2.9.x"
+    "@vitejs/plugin-react": "^3.0.1",
+    "vite": "4.0.x"
   }
 }

--- a/examples/vanilla/package.json
+++ b/examples/vanilla/package.json
@@ -8,6 +8,6 @@
     "serve:node": "node ../node-server/index vanilla"
   },
   "dependencies": {
-    "vite": "2.9.x"
+    "vite": "4.0.x"
   }
 }

--- a/examples/vue/package.json
+++ b/examples/vue/package.json
@@ -8,11 +8,11 @@
     "serve:node": "node ../node-server/index vue"
   },
   "dependencies": {
-    "@vitejs/plugin-vue": "^2.0.1",
+    "@vitejs/plugin-vue": "^4.0.0",
     "@vue/compiler-sfc": "^3.2.26",
     "@vue/server-renderer": "^3.2.26",
     "@vueuse/head": "^0.6.0",
-    "vite": "2.9.x",
+    "vite": "4.0.x",
     "vue": "^3.2.26",
     "vue-router": "^4.0.12"
   }

--- a/package.json
+++ b/package.json
@@ -45,14 +45,14 @@
     "test/fixtures/*"
   ],
   "peerDependencies": {
-    "@vitejs/plugin-react": "^1.1.3",
-    "@vitejs/plugin-vue": "^2.0.1",
+    "@vitejs/plugin-react": "^3.0.1",
+    "@vitejs/plugin-vue": "^4.0.0",
     "@vueuse/head": "0.x",
     "react": "^17",
     "react-dom": "^17",
     "react-helmet-async": "^1.0.0",
     "react-router-dom": "^6",
-    "vite": "^2.7.7",
+    "vite": "^4.0.4",
     "vue": "^3.2.26",
     "vue-router": "^4.0.0"
   },

--- a/src/dev/server.ts
+++ b/src/dev/server.ts
@@ -31,8 +31,17 @@ export const createSSRDevHandler = (
   server: ViteDevServer,
   options: SsrOptions = {}
 ) => {
+  // Fix, because newer versions of vite's inlineConfig contains "ssr" property which type is object
+  let { ssr, ...otherOptions } = server.config.inlineConfig as InlineConfig & {
+    ssr?: InlineConfig['ssr'] | string
+  }
+
+  if (typeof ssr != 'string') {
+    ssr = undefined
+  }
+
   options = {
-    ...server.config.inlineConfig, // CLI flags
+    ...{ ssr, ...otherOptions }, // CLI flags
     ...options,
   }
 

--- a/test/fixtures/react-ts-basic/package.json
+++ b/test/fixtures/react-ts-basic/package.json
@@ -15,8 +15,8 @@
   "devDependencies": {
     "@types/connect": "^3.4.35",
     "@types/node": "^16.4.7",
-    "@vitejs/plugin-react": "^1.1.3",
+    "@vitejs/plugin-react": "^3.0.1",
     "typescript": "^4.3.2",
-    "vite": "2.9.x"
+    "vite": "4.0.x"
   }
 }

--- a/test/fixtures/vue-ts-basic/package.json
+++ b/test/fixtures/vue-ts-basic/package.json
@@ -14,10 +14,10 @@
   "devDependencies": {
     "@types/connect": "^3.4.35",
     "@types/node": "^16.4.7",
-    "@vitejs/plugin-vue": "^2.0.1",
+    "@vitejs/plugin-vue": "^4.0.0",
     "@vue/compiler-sfc": "^3.2.26",
     "typescript": "^4.3.2",
-    "vite": "2.9.x",
+    "vite": "4.0.x",
     "vue-tsc": "^0.0.24"
   }
 }

